From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Serin Yeh <serin.yeh@intel.com>
Date: Fri, 24 Apr 2026 10:19:13 +0800
Subject: [PATCH] media: ipu-bridge: Add support additional link frequency

Support additional frequency 900 MHz for the OV05C10 sensor

Signed-off-by: Serin Yeh <serin.yeh@intel.com>
Reviewed-by: Hans de Goede <johannes.goede@oss.qualcomm.com>
Signed-off-by: Sakari Ailus <sakari.ailus@linux.intel.com>
(cherry picked from commit f9536a8065d9eed0d9114878c56ffc4e5a18bdfc)
Link: https://git.kernel.org/pub/scm/linux/kernel/git/next/linux-next.git/commit/?id=f9536a8065d9eed0d9114878c56ffc4e5a18bdfc
[mdick85: queued in media-next for v7.4; the ov05c10 on the HP EliteBook
 X G2i needs the 900 MHz link or it fails to probe. Drop at the next
 base bump that contains it. Note that b883954d842f ("media: ipu-bridge:
 Match sensor configs per IPU and add config flags", also v7.4)
 restructures ipu_supported_sensors[], so this hunk will not apply to a
 v7.4 base as is.]
Signed-off-by: Meir Dick <mdick85@gmail.com>
---
 drivers/media/pci/intel/ipu-bridge.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/drivers/media/pci/intel/ipu-bridge.c b/drivers/media/pci/intel/ipu-bridge.c
index b1e57d2..b0c2ca0 100644
--- a/drivers/media/pci/intel/ipu-bridge.c
+++ b/drivers/media/pci/intel/ipu-bridge.c
@@ -81,7 +81,7 @@ static const struct ipu_sensor_config ipu_supported_sensors[] = {
 	/* Omnivision OV02E10 */
 	IPU_SENSOR_CONFIG("OVTI02E1", 1, 360000000),
 	/* Omnivision ov05c10 */
-	IPU_SENSOR_CONFIG("OVTI05C1", 1, 480000000),
+	IPU_SENSOR_CONFIG("OVTI05C1", 2, 480000000, 900000000),
 	/* Omnivision OV08A10 */
 	IPU_SENSOR_CONFIG("OVTI08A1", 1, 500000000),
 	/* Omnivision OV08x40 */
